language/move-lang/src/expansion/ast.rs

```diff
@@ -96,7 +96,14 @@ pub struct Function {
 // Specification Blocks
 //**************************************************************************************************
 
-// Specification block:
+#[derive(Debug, PartialEq)]
+pub struct FunctionPattern_ {
```

Comment:
Maybe `SpecFunctionPattern_`? Patterns are another concept in the language. While they aren't explicitly mentioned in the `lvalue` ast, they might be in errors. So I think indicating this is for some sort of spec apply thingy is useful

Reply:
Renamed this to SpecApplyPattern.